packages/sdk/src/compound-processor.ts

```diff
@@ -27,29 +27,36 @@ export class CompoundSpanProcessor extends BatchSpanProcessor {
   constructor(
     exporter: SpanExporter,
     private readonly processors: SpanProcessor[],
+    private readonly logger: DiagLogger,
   ) {
     super(exporter)
-    this.processors = processors
   }
 
   override onStart(span: Span, parentContext: Context): void {
+    this.logger.verbose("span start", span, parentContext)
     super.onStart(span, parentContext)
     this.processors.forEach((p) => {
       p.onStart(span, parentContext)
     })
   }
 
   override onEnd(span: ReadableSpan): void {
-    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    this.logger.verbose("span end", span)
     ;[...this.processors].reverse().forEach((p) => {
       p.onEnd(span)
     })
     super.onEnd(span)
   }
 
   override async forceFlush(): Promise<void> {
-    await Promise.all(this.processors.map((p) => p.forceFlush()))
-    await super.forceFlush()
+    this.logger.debug("flush")
+    try {
+      await Promise.all(this.processors.map((p) => p.forceFlush()))
+      await super.forceFlush()
+    } catch (error) {
+      this.logger.error("flush error", error)
+      throw error
+    }
```

Comment:
This does not run unless it's inside a try/catch when instrumenting through the lambda instrumentation (as discussed during standup). I'm assuming it's triggering some extreme edge case in V8 and it gets miscompiled.